inc/checks.php

```diff
@@ -0,0 +1,247 @@
+	}
+
+	$values['standard'][] = NS_THEME_CHECK_DIR . '/bin/phpcs.xml';
+
+	// Ignore unrelated files from the check.
+	$values['ignored'] = array(
+		'.*/node_modules/.*',
+	);
+
+	ob_start();
+	$num_errors = $phpcs_cli->process( $values );
+	$raw_output = ob_get_clean();
+
+	// Sniff theme files.
+	if ( 1 === absint( $args['raw_output'] ) ) {
+		echo '<pre>' . esc_html( $raw_output ) . '</pre>';
+	} else {
+		$output = json_decode( $raw_output );
+		if ( ! empty( $output ) ) {
+			ns_theme_check_render_json_report( $output );
+		}
+	}
+
+	// Has the theme passed?
+	if ( 0 === $num_errors ) {
+		return true;
+	} else {
+		return false;
+	}
+
+}
+
+/**
+ * Perform style.css header check.
+ *
+ * @since 0.3.0
+ *
+ * @param string $theme_slug Theme slug.
+ * @param array  $theme WP_Theme Theme object.
+ *
+ * @return bool
+ */
+function ns_theme_check_style_headers( $theme_slug, $theme ) {
+
+	$pass             = true;
+	$required_headers = array(
+		'Name',
+		'Description',
+		'Author',
+		'Version',
+		// @see https://core.trac.wordpress.org/ticket/40034
+		// 'License',
+		// 'License URI',
+		'TextDomain',
+	);
+
+	foreach ( $required_headers as $header ) {
+		if ( $theme->get( $header ) ) {
+			continue;
+		}
+		$notices[] = array(
+			/* translators: 1: comment header line, 2: style.css */
+			'message'  => sprintf(
+				__( 'The %1$s is not defined in the style.css header.', 'ns-theme-check' ),
+				$header
+			),
+			'severity' => 'error',
+		);
+	}
+
+	if ( strpos( $theme_slug, 'wordpress' ) || strpos( $theme_slug, 'theme' ) ) {
+		$notices[] = array(
+			'message'  => __( 'The theme name cannot contain WordPress or Theme.', 'ns-theme-check' ),
+			'severity' => 'error',
+		);
+	}
+
+	if ( preg_match( '|[^\d\.]|', $theme->get( 'Version' ) ) ) {
+		$notices[] = array(
+			'message' => __( 'Version strings can only contain numeric and period characters (like 1.2).', 'ns-theme-check' ),
+			'severity' => 'error',
+		);
+	}
+
+	// Prevent duplicate URLs.
+	$themeuri  = trim( $theme->get( 'ThemeURI' ) , '/\\' );
+	$authoruri = trim( $theme->get( 'AuthorURI' ) , '/\\' );
+	if ( $themeuri === $authoruri ) {
+		$notices[] = array(
+			'message'  => __( 'Duplicate theme and author URLs. A theme URL is a page/site that provides details about this specific theme. An author URL is a page/site that provides information about the author of the theme. The theme and author URL are optional.', 'ns-theme-check' ),
+			'severity' => 'error',
+		);
+	}
+
+	if ( $theme_slug === $theme->get( 'Text Domain' ) ) {
+		$notices[] = array(
+			/* translators: %1$s: Text Domain, %2$s: Theme Slug */
+			'message' => sprintf( __( 'The text domain "%1$s" must match the theme slug "%2$s".', 'ns-theme-check' ),
+				$theme->get( 'TextDomain' ),
+				$theme_slug
+			),
+			'severity' => 'error',
+		);
+	}
+
+	$registered_tags    = ns_theme_check_get_theme_tags();
+	$tags               = array_map( 'strtolower', $theme->get('Tags') );
```

Comment:
Travis is failing due to this line. `Expected 1 spaces after opening bracket; 0 found`